### PR TITLE
Fix JSONDecodeError in Datafusion operators

### PR DESCRIPTION
**Pull request by VladaZakharova**
---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: airflow/providers/google/cloud/hooks/datafusion.py [hunk 1]
```diff
@@ -339,6 +341,8 @@ def delete_pipeline(
         response = self._cdap_request(url=url, method="DELETE", body=None)
         if response.status != 200:
             raise AirflowException(f"Deleting a pipeline failed with code {response.status}")
+        if response.data is None:
+            raise AirflowException("Deleting a pipeline failed because of empty data in received response")
```
Comment (bhirsz) [inline airflow/providers/google/cloud/hooks/datafusion.py:346]: It's fine in general though we have a lot of repeated code - ie get/delete/create operations have all != 200 and None checks.

File: airflow/providers/google/cloud/hooks/datafusion.py [hunk 2]
```diff
@@ -339,6 +341,8 @@ def delete_pipeline(
...
```
Comment (lwyszomi) [inline airflow/providers/google/cloud/hooks/datafusion.py:345]: Update the message: `Empty response received. Please, check for possible root causes of this behavior either in DAG code or on Cloud Datafusion side`

File: airflow/providers/google/cloud/hooks/datafusion.py (hunk 1 above)
Comment (VladaZakharova) [reply airflow/providers/google/cloud/hooks/datafusion.py:346]: Yes, i will try to use another approach to decrease duplication in the code
Thanks!

File: airflow/providers/google/cloud/hooks/datafusion.py (hunk 2 above)
Comment (VladaZakharova) [reply airflow/providers/google/cloud/hooks/datafusion.py:345]: Yes, this message sounds more appropriate :)

File: airflow/providers/google/cloud/hooks/datafusion.py [hunk 3]
```diff
@@ -387,8 +398,9 @@ def get_pipeline_workflow(
             quote(pipeline_id),
         )
         response = self._cdap_request(url=url, method="GET")
-        if response.status != 200:
-            raise AirflowException(f"Retrieving a pipeline state failed with code {response.status}")
+        self._check_response_status_and_data(
+            response, f"Retrieving a pipeline state failed with " f"code {response.status}"
```
Comment (bhirsz) [inline airflow/providers/google/cloud/hooks/datafusion.py:402]: it looks like you split string manually and then black moved it to one line. There is " f" now

Comment (VladaZakharova) [reply airflow/providers/google/cloud/hooks/datafusion.py:402]: yes, i know
i am fixing now it :)

**Review by lwyszomi — APPROVED** [review]
I think we are ready to push this to the community

**Pull request closed without merging**